src/BlazorUI/Bit.BlazorUI.Tests/Buttons/BitButtonTests.cs

```diff
@@ -174,6 +175,33 @@ public void BitButtonTypeOfButtonTest(BitButtonType buttonType)
         var bitButton = com.Find(".bit-btn");
 
         var buttonTypeName = buttonType == BitButtonType.Button ? "button" : buttonType == BitButtonType.Submit ? "submit" : "reset";
-        Assert.AreEqual(bitButton.GetAttribute("type"), buttonTypeName);
+        Assert.AreEqual(buttonTypeName, bitButton.GetAttribute("type"));
     }
-}
+
+    [TestMethod]
+    public void BitButtonSubmitStateInEditContextTest()
+    {
+        var com = RenderComponent<BitButton>(parameters =>
```

Comment:
I explicitly did not test against `BitButtonTest` as directly testing is way easier.

In theory (in a follow up PR) we could remove this `BitButtonTest` and friends completely. The only reason is to have a `CascadingParameter` which we could add via `p => p.AddCascadingValue` anyway.

Reply:
yeah, you are right about these ...Test.razor component completely.
we kept adding these components as a good practice for future requirements (if any come up), but for now, we don't have any specific use for them.
although these middleware components are creating some issues/bugs testing 2-way parameters. let's discuss it in another thread.